packages/candid/src/idl.ts

```diff
@@ -861,7 +861,11 @@ export class RecordClass extends ConstructType<Record<string, any>> {
       idx++;
     }
     for (const [expectKey, expectType] of this._fields.slice(idx)) {
-      if (expectType instanceof OptClass) {
+      if (
+        expectType instanceof OptClass ||
+        expectType instanceof ReservedClass ||
+        expectType instanceof NullClass
```

Comment:
I think `null` is actually wrong here. This PR is trying to implement this rule, I assume:

```
<nat> not in <fieldtype>;*
record { <fieldtype>;* } <: record { <fieldtype'>;* }
opt empty <: <datatype'>
------------------------------------------------------------------------------
record { <fieldtype>;* } <: record { <nat> : <datatype'>; <fieldtype'>;* }
```

note that the lower bound is `opt empty`, not `null`, so only `opt t` and `reserved` are included. (I’m not saying it would be morally wrong to treat `null` that way, but it’s not what the spec says.)

Reply:
Right. Fixed.